### ISSUE TITLE
Foreign_key Error in rake db:rollback, ruby-ibmdb 2.6.1

### ISSUE THREAD

**Issue by zjuanma** [ibmdb/ruby-ibmdb#37]
Hi,
Rake db:rollback not work when we have migrations with add_foreign_key.

```ruby
 class CreateLawyerOfficeAreas < ActiveRecord::Migration
  def change
    create_table :lawyer_office_areas do |t|
      t.references :lawyer_office, null: false, index: true
      t.references :province, null: false, index: true
      t.references :judicial_district, index: true
      t.timestamps
    end

    add_foreign_key :lawyer_office_areas, :lawyer_offices, on_delete: :cascade
    add_foreign_key :lawyer_office_areas, :provinces, on_delete: :cascade
    add_foreign_key :lawyer_office_areas, :judicial_districts, on_delete: :cascade

  end
end
```

```
user1000@5e0cee1c1941:~$ bundle exec rake db:migrate
== 20150721102932 CreateLawyerOfficeAreas: migrating ==========================
-- create_table(:lawyer_office_areas)
   -> 0.2923s
-- add_foreign_key(:lawyer_office_areas, :lawyer_offices, {:on_delete=>:cascade})
   -> 0.0110s
-- add_foreign_key(:lawyer_office_areas, :provinces, {:on_delete=>:cascade})
   -> 0.0142s
-- add_foreign_key(:lawyer_office_areas, :judicial_districts, {:on_delete=>:cascade})
   -> 0.0281s
== 20150721102932 CreateLawyerOfficeAreas: migrated (0.3464s) =================


user1000@5e0cee1c1941:~$ bundle exec rake db:rollback
== 20150721102932 CreateLawyerOfficeAreas: reverting ==========================
-- remove_foreign_key(:lawyer_office_areas, :judicial_districts)
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

Table 'lawyer_office_areas' has no foreign key on column 'judicial_district_id'/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/connection_adapters/abstract/schema_statements.rb:780:in `block in remove_foreign_key'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/connection_adapters/abstract/schema_statements.rb:774:in `fetch'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/connection_adapters/abstract/schema_statements.rb:774:in `remove_foreign_key'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:662:in `block in method_missing'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:632:in `block in say_with_time'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:632:in `say_with_time'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:652:in `method_missing'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:497:in `block in revert'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:496:in `each'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:496:in `revert'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:604:in `exec_migration'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:590:in `block (2 levels) in migrate'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:589:in `block in migrate'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:292:in `with_connection'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:588:in `migrate'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:765:in `migrate'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:995:in `block in execute_migration_in_transaction'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:1041:in `block in ddl_transaction'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `block in transaction'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/connection_adapters/abstract/transaction.rb:188:in `within_new_transaction'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `transaction'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/transactions.rb:220:in `transaction'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:1041:in `ddl_transaction'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:994:in `execute_migration_in_transaction'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:956:in `block in migrate'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:952:in `each'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:952:in `migrate'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:827:in `down'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:907:in `move'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:809:in `rollback'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/railties/databases.rake:125:in `block (2 levels) in <top (required)>'
ArgumentError: Table 'lawyer_office_areas' has no foreign key on column 'judicial_district_id'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/connection_adapters/abstract/schema_statements.rb:780:in `block in remove_foreign_key'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/connection_adapters/abstract/schema_statements.rb:774:in `fetch'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/connection_adapters/abstract/schema_statements.rb:774:in `remove_foreign_key'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:662:in `block in method_missing'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:632:in `block in say_with_time'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:632:in `say_with_time'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:652:in `method_missing'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:497:in `block in revert'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:496:in `each'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:496:in `revert'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:604:in `exec_migration'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:590:in `block (2 levels) in migrate'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:589:in `block in migrate'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:292:in `with_connection'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:588:in `migrate'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:765:in `migrate'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:995:in `block in execute_migration_in_transaction'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:1041:in `block in ddl_transaction'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `block in transaction'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/connection_adapters/abstract/transaction.rb:188:in `within_new_transaction'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `transaction'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/transactions.rb:220:in `transaction'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:1041:in `ddl_transaction'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:994:in `execute_migration_in_transaction'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:956:in `block in migrate'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:952:in `each'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:952:in `migrate'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:827:in `down'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:907:in `move'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/migration.rb:809:in `rollback'
/ruby_gems/2.1/gems/activerecord-4.2.1/lib/active_record/railties/databases.rake:125:in `block (2 levels) in <top (required)>'
Tasks: TOP => db:rollback
(See full trace by running task with --trace)

```

**Comment by zjuanma** [comment]
In this line

https://github.com/ibmdb/ruby-ibmdb/blob/master/IBM_DB_Adapter/ibm_db/lib/active_record/connection_adapters/ibm_db_adapter.rb#L1954

 IBM_DB.fetch_array(stmt) returns false

**Comment by zjuanma** [comment]
Temporal patch to solved this issue:

```
module ActiveRecord
  module ConnectionAdapters
    class IBM_DBAdapter
      alias_method :foreign_keys_original, :foreign_keys
      def foreign_keys(table_name)
        foreign_keys = []
        query = "select constname, fk_colnames, reftabname, pk_colnames from SYSCAT.REFERENCES where tabname = '#{table_name.upcase}'"
        stmt = execute(query)

        while ( fk_row = IBM_DB.fetch_array(stmt) )
          fk_row.map!(&:strip).map!(&:downcase)
          options = {
            name: fk_row[0],
            column: fk_row[1],
            primary_key: fk_row[3],
          }
          foreign_keys << ForeignKeyDefinition.new(table_name, fk_row[2], options)
        end

        foreign_keys
      end
    end
  end
end
```

**Comment by arvindgu** [comment]
We have released ibm_db gem version 3.0.0 and fixed this issue.

**Comment by arvindgu** [comment]
Can this be closed?
